src/conditional.ts

```diff
@@ -16,3 +16,26 @@ export abstract class Conditional<T extends Step> {
    */
   abstract accept(): boolean | Promise<boolean>;
 }
+
+export async function getAndCacheDependency(
```

Comment:
Moved this out of `sortedSteps` to reuse in `walker.ts`

Reply:
No need to move this anymore since it new uses `sortedSteps`.